### ISSUE TITLE
[Filebeat-7.x] Mapping issue breaks Kibana monitoring and Filebeat built-in modules 

### ISSUE THREAD

**Issue by jesusgn90** [wazuh/wazuh#3723]
## Description

> Wazuh 3.9.3 - Elastic stack 7.2.0

- X-Pack monitoring app not working in Kibana due to mapping issues related to Filebeat indices.
- Filebeat built-in modules are not working due to mapping issues too.

## Why?

Filebeat is expected to insert its template. This template is composed dynamically using `fields.yml` file which
is also composed dynamically using the `fields.yml` from all the built-in modules.

All built-in modules send their data to `filebeat-7.2.0-*` indices as is described in the Filebeat template:

```js
"index_patterns": [
  "filebeat-7.2.0-*"
],
```

We are using the next lines as global configuration for Filebeat:

```yaml
setup.template.json.enabled: true
setup.template.json.path: '/etc/filebeat/wazuh-template.json'
setup.template.json.name: 'wazuh'
setup.template.overwrite: true
```

What we are doing is an overwrite of the Filebeat template, then both Filebeat and its built-in modules have no valid template.

This is not a problem for Wazuh itself but for Filebeat, its modules and the monitoring app in Kibana it does.

#### Why is it a problem for Kibana monitoring?

Because Kibana executes some queries against the `filebeat*` indices, causing failures due to mapping issues.

The Filebeat monitoring itself can work properly, it stores the data in `.monitoring-beats*` indices but the Kibana monitoring app
looks for more indices.

#### Why is it a problem for Filebeat modules?

We are not loading the Filebeat template, so every `filebeat*` index is corrupt. All Filebeat modules send the data to `filebeat*` indices,
so their built-in dashboards are not working properly because the expected mapping for their fields differ from how it is actually.

### Things to be kept in mind

1. We can't merge both templates, the Filebeat template is dynamic and it's about x10 times bigger than ours. Also, we may have problems with index related settings and a few other important things.
2. `setup.template.*` blocks are unique and the last one takes precedence over the previous block, so just one block is valid.
3. You can't send more than one template at once using Filebeat.
4. Modules _can't_ send templates (instead, they can add its fields to the general Filebeat template using fields.yml files, we'll talk about it later).

## Proposals

First of all, I'm going to show a few of `filebeat` commands where `-E` is an in-line setting applied just for running the command. They have precedence against the configuration files.

**1. Insert the Filebeat template manually, using the `filebeat` CLI before starting the service:**

```
filebeat setup --index-management -E setup.template.json.enabled=false -E 'output.elasticsearch.hosts=["172.16.1.2:9200"]'
```

Where `setup.template.json.enabled=false` tells Filebeat to ignore our `setup.template.json.*` lines, then it will load its template.

Pros:
- All thins will work as expected.

Cons:
- This would be an **extra** step in the installation guide before starting the Filebeat service.

**2. Insert the Wazuh template manually, using the `filebeat` CLI and removing all `setup.template.*` lines from our configuration.**

```
filebeat setup --index-management -E setup.template.json.enabled=true -E setup.template.json.path=/etc/filebeat/wazuh-template.json -E setup.template.json.name=wazuh -E 'output.elasticsearch.hosts=["172.16.1.2:9200"]'
```

Details about the command:
- `setup.template.json.enabled=true` tells Filebeat we want to use a JSON template.
- `setup.template.json.path=/etc/filebeat/wazuh-template.json` tells Filebeat where to look for the template.
- `setup.template.json.name=wazuh` tells Filebeat which name we want for our template once it's stored.
- `output.elasticsearch.hosts=["172.16.1.2:9200"]` tells Filebeat where to send the template.

Pros:
- All things will work as expected too.
- The configuration file `/etc/filebeat/filebeat.yml` would be reduced because we no longer need the `setup.template.*` lines.

Cons:
- This would be an **extra** step in the installation guide before starting the Filebeat service.
- We are in risk of human-related errors, causing Wazuh alert indices being created with no mapping if a user starts Filebeat before doing this step.

**3. Make the Wazuh app insert the alerts template.**

Pros:
- All things will work as expected too.
- The configuration file `/etc/filebeat/filebeat.yml` would be reduced because we no longer need the `setup.template.*` lines.
Cons:
- You are expecting the Wazuh app is installed, some users may be using the stack with no app.
- Alerts must come once the app is installed.
- We are in risk of human-related errors, causing Wazuh alert indices being created with no mapping if a user starts Filebeat before doing this step.

**4. Using the `fields.yml` file properly configured.**

Pros:
- Simplifies the installation guide, all would come with the module.
- No longer needed to maintain the JSON template, instead, we would maintain the `fields.yml`
- This would be the right way.
Cons:
- If we want the perfect module, it should be merged to Beats repository.
- I don't know how to index settings such as index replicas or that kind of settings would affect other modules. I think it would be easily configurable, but I didn't try it before.
- Dynamic fields, how they work with this kind of YML templates? they might work with no problems but I didn't try it before.

**5. (Really it would be 4-b) Simplify our fields, follow the ECS standard.**

Field alias, ECS, all the normalization would be nice for a future Wazuh. Making our product fully compatible with other components from the Elastic stack and doing all as they are doing.

Pros:
- Following best practices
- Drastically reduced mapping
Cons:
- Breaking change, no retro-compatibility
- Bigger efforts than other solutions
- All the _cons_ from solution `#4`

## Conclusion

- We still need to find a proper solution for this.
- The Elastic team is also reviewing the use case, maybe they give us some light but as I can see in some other forums and threads from the Elastic discuss forum, it's a problem with no easy solution.
- From the above solutions, I would say to use `1.` (insert the Filebeat template manually using `filebeat` CLI).

**Comment by jesusgn90** [comment]
**Update**

Here is a PR https://github.com/wazuh/wazuh/pull/3727 which covers the Wazuh integration and fixes the Kibana Monitoring issue.

**Comment by moshe0076** [comment]
Hi,

I see the same issue using wazuh:3.10.2_7.3.2 Docker and elastic stack 7.3.2

Thanks!
